cmd/iam-store.go

```diff
@@ -1448,33 +1448,35 @@ func (store *IAMStoreSys) DeleteUsers(ctx context.Context, users []string) error
 	return nil
 }
 
+// ParentUserInfo represents some general information about a user's parentUser.
+// It contains the parentUserid as well as the sessiontoken with which the child
+// user was created
+type ParentUserInfo struct {
+	ParentUser   string
+	SessionToken string
+}
+
 // GetAllParentUsers - returns all distinct "parent-users" associated with STS or service
 // credentials.
-func (store *IAMStoreSys) GetAllParentUsers() []string {
+func (store *IAMStoreSys) GetAllParentUsers() []ParentUserInfo {
 	cache := store.rlock()
 	defer store.runlock()
 
-	res := set.NewStringSet()
+	parentUserSessionTokenMap := map[string]string{}
 	for _, cred := range cache.iamUsersMap {
 		if cred.IsServiceAccount() || cred.IsTemp() {
-			parentUser := cred.ParentUser
-			if cred.SessionToken != "" {
-				claims, err := getClaimsFromToken(cred.SessionToken)
-				if err != nil {
-					continue
-				}
-				if v, ok := claims[subClaim]; ok {
-					subFromToken, ok := v.(string)
-					if ok {
-						parentUser = subFromToken
-					}
-				}
-			}
-			res.Add(parentUser)
+			parentUserSessionTokenMap[cred.ParentUser] = cred.SessionToken
```

Comment:
This looks completely wrong cred.ParentUser is not meaningful in any manner here - you are returning way too many entries here to expire only a sub-set of entries. 

The previous code actually correctly handles this.

Reply:
In theory the method returns the same set of entries as before (since we are using a Map). The change here is that the method returns a `ParentUserInfo` struct instead of plain userid, which is required in order to extract OIDC user id when fetching user information at some later point (`LookupUser`).